aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -434,12 +437,15 @@ Tensor& index_add_cpu_(Tensor & self, int64_t dim, const Tensor & index, const T
       // TODO: Maybe TensorAccessor can beused here?
       auto* self_ptr = self.data_ptr<scalar_t>();
       auto* source_ptr = source.data_ptr<scalar_t>();
-      for (auto i = 0; i < numel; i++) {
-        auto self_i = index_data[i];
-        TORCH_CHECK_INDEX((self_i >= 0) && (self_i < self.numel()), "index out of range in self");
-        scalar_t *self_ip = self_ptr + self_i * self_stride;
-        *self_ip += *(source_ptr + i * source_stride);
-      }
+      AT_DISPATCH_INDEX_TYPES(index.scalar_type(), "index_add_cpu_", [&] () {
```

Comment:
I'm hitting some windows build failures here - https://app.circleci.com/pipelines/github/pytorch/pytorch/233453/workflows/e2a13402-bc72-474f-89a7-6c1f0132cdd0/jobs/8597430/parallel-runs/0/steps/0-104

Seems variables are not captured correctly in the nested lambda. Also found some asks about similar issues online - https://developercommunity.visualstudio.com/content/problem/1150322/nested-c20-template-lambdas-two-issues-with-captur.html

Hi @peterjc123 ! @ngimel suggested that you might have solution/workaround to this issue. Would you mind taking a look? Thank you in advance!

Reply:
Explicitly capturing variables could work around it.